templates/default/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+//! Note: This contract is not meant for production, it is used to illustrate
+//! how to use the standard library and the tooling Concordium provides. There
+//! is no claim that the logic of the contract is reasonable, or safe.
+//!
+//! # A Concordium V1 smart contract
+use concordium_std::*;
+
+/// Your smart contract state.
+#[derive(Serialize, SchemaType)]
+pub struct State {
+    // Your state
+}
+
+/// Your smart contract errors.
+#[derive(Reject, Serial, SchemaType)]
+enum Error {
+    /// YourError
+    YourError,
+}
+
+/// Init function that creates a new smart contract.
+#[init(contract = "{{crate_name}}")]
+fn your_smart_contract_init<S: HasStateApi>(
+    _ctx: &impl HasInitContext,
+    _state_builder: &mut StateBuilder<S>,
+) -> InitResult<State> {
+    // Your code
+
+    Ok(State {})
+}
+
+/// Payable Receive function.
+#[receive(contract = "{{crate_name}}", name = "receiveFunction", payable, mutable, error = "Error")]
+fn your_smart_contract_receive_function<S: HasStateApi>(
+    _ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+    _amount: Amount,
+) -> Result<(), Error> {
+    // Your code
+
+    let error_occurred = false;
+
+    if error_occurred {
+        Err(Error::YourError)
+    } else {
+        Ok(())
+    }
```

Comment:
I think it would be good to include a parameter here. For example something like this? `42` was chosen due to a lack of imagination. Please use something else if you have a better idea.

```suggestion
    let input = ctx.parameter_cursor().get()?; // Returns Error::ParseError on failure

    if input == 42 {
        Ok(())
    } else {
        Err(Error::YourError)
    }
```

**Then we could also show how to set a parameter in a test of the receive function (and construct a `TestStateHost`).**

Reply:
The input parameter is now the boolean variable `throw_error`. If `true` the receive function will throw a custom error, if `false` the receive function executes successfully.